--- a/hyprlock.conf
+++ b/hyprlock.conf
@@ -30,9 +30,9 @@ label {
     valign = top
 }
 
-# DATE 
+# DATE
 label {
-    monitor = 
+    monitor =
     text = cmd[update:43200000] echo "$(date +"%A, %d %B %Y")"
     color = $text
     font_size = 25
@@ -45,8 +45,8 @@ label {
 # USER AVATAR
 
 image {
-    monitor = 
-    path = ~/.face
+    monitor =
+    path = ~/.config/user-icon.jpg
     size = 100
     border_color = $accent
 
@@ -76,4 +76,4 @@ input-field {
     position = 0, -35
     halign = center
     valign = center
-}
\ No newline at end of file
+}
